### PR TITLE
Bump Prometheus and Prometheus Operator images and update manifests

### PR DESCRIPTION
**Pull request by oprinmarius**
**What type of PR is this?**

/kind feature
/kind design

**What this PR does / why we need it**:

* Updates Prometheus from version 0.46.0 to version 0.55.0
* Update Prometheus Operator from v2.25.0 to v2.34.0

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
part of https://github.com/kubernetes/perf-tests/issues/2035
**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: clusterloader2/pkg/prometheus/manifests/0ssd-storage-class.yaml [hunk 1]
```diff
@@ -7,7 +7,18 @@ metadata:
   name: ssd
 provisioner: {{$PROMETHEUS_STORAGE_CLASS_PROVISIONER}}
 parameters:
+  {{if eq $PROMETHEUS_STORAGE_CLASS_PROVISIONER "kubernetes.io/gce-pd"}}
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/0ssd-storage-class.yaml:10]: It's not directly related to image bump. Please open separate PR (which makes potential rollback easier).

File: clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml [hunk 2]
```diff
@@ -46,7 +49,36 @@ spec:
...
+          capabilities:
+            drop:
+            - ALL
           readOnlyRootFilesystem: true
+      - args:
+        - --logtostderr
+        - --secure-listen-address=:8443
+        - --tls-cipher-suites=TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
+        - --upstream=http://127.0.0.1:8080/
+        image: quay.io/brancz/kube-rbac-proxy:v0.11.0
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml:61]: Please use gcr.io/k8s-testimages/quay.io/brancz/kube-rbac-proxy:v0.11.0

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml [hunk 3]
```diff
@@ -6,15 +6,36 @@ apiVersion: monitoring.coreos.com/v1
...
-  baseImage: gcr.io/k8s-testimages/quay.io/prometheus/prometheus
+  alerting:
+    alertmanagers:
+    - apiVersion: v2
+      name: alertmanager-main
+      namespace: monitoring
+      port: web
+  enableFeatures: []
+  externalLabels: {}
+  image: quay.io/prometheus/prometheus:v2.34.0
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:25]: Please use gcr.io/k8s-testimages/quay.io/prometheus/prometheus:v2.34.0

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml [hunk 4]
```diff
@@ -32,36 +53,13 @@ spec:
...
-    volumeClaimTemplate:
-      spec:
-        storageClassName: ssd
-        resources:
-          requests:
-            # Start with 10Gi, add 10Gi for each 1K nodes.
-            storage: {{MultiplyInt 10 (AddInt 1 (DivideInt .Nodes 1000))}}Gi
-  query:
-    maxSamples: 100000000
-  {{if $PROMETHEUS_SCRAPE_WINDOWS_NODES}}
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:63]: We do need to support PROMETHEUS_SCRAPE_WINDOWS_NODES flag.

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml [hunk 5]
```diff
@@ -32,36 +53,13 @@ spec:
...
-  version: v2.25.0
-  retention: 7d
-  storage:
-    volumeClaimTemplate:
-      spec:
-        storageClassName: ssd
-        resources:
-          requests:
-            # Start with 10Gi, add 10Gi for each 1K nodes.
-            storage: {{MultiplyInt 10 (AddInt 1 (DivideInt .Nodes 1000))}}Gi
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:60]: Please do not remove the storage sizing logic.

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml [hunk 6]
```diff
@@ -32,36 +53,13 @@ spec:
...
-  query:
-    maxSamples: 100000000
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:62]: Please keep maxSamples.

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml [hunk 7]
```diff
@@ -6,15 +6,36 @@ apiVersion: monitoring.coreos.com/v1
 kind: Prometheus
```
Comment (mborsz) [inline clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:6]: It looks like this file drops overrides some existing configurations used by our tests. I added few comments, but I think there are also other cases that should be rolled back.

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml (hunk 5 above)
Comment (oprinmarius) [reply clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:60]: https://github.com/kubernetes/perf-tests/pull/2040/commits/ec88962d79cc233adfb94807b34d5d6bfe591d91
This commit adds this back , and it's part of this PR.  I will update this file after I open a separate PR for Azure-Disk support

File: clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml (hunk 6 above)
Comment (oprinmarius) [reply clusterloader2/pkg/prometheus/manifests/prometheus-prometheus.yaml:62]: https://github.com/kubernetes/perf-tests/pull/2040/commits/ec88962d79cc233adfb94807b34d5d6bfe591d91
This commit adds this back , and it's part of this PR.  I will update this file after I open a separate PR for Azure-Disk support

File: clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml [hunk 8]
```diff
@@ -19,20 +19,23 @@ spec:
...
-        app.kubernetes.io/version: 0.46.0
+        app.kubernetes.io/version: 0.55.0
     spec:
+      automountServiceAccountToken: true
       containers:
       - args:
         {{if $PROMETHEUS_SCRAPE_KUBELETS}}
         - --kubelet-service=kube-system/kubelet
         {{end}}
-        - --prometheus-config-reloader=gcr.io/k8s-testimages/quay.io/prometheus-operator/prometheus-config-reloader:v0.46.0
```
[the remark this answers is not in the document]
Comment (mborsz) [reply clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml:33]: Hmm, weird. I'm sure I pushed both images above and validated that they are present now.

Could you paste exact image path you are trying to use and/or some error you are seeing?

[the remark this answers is not in the document]
Comment (oprinmarius) [reply clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml:33]: The images are there now, they weren't present before I opened the PR, and I used images without gcr.io/k8s-testimages prefix for testing

**Pull request closed without merging**